malariagen_data/ag3.py

```diff
@@ -341,3 +350,78 @@ def _results_cache_add_analysis_params(self, params):
         super()._results_cache_add_analysis_params(params)
         # override parent class to add AIM analysis
         params["aim_analysis"] = self._aim_analysis
+
+    def load_inversion_tags(self, inversion: str):
+        # needs to be modified depending on where we are hosting
+        import importlib.resources
+        from . import resources
+
+        with importlib.resources.path(resources, "karyotype_tag_snps.csv") as path:
+            df_tag_snps = pd.read_csv(path, sep=",")
+        return df_tag_snps.query(f"inversion == '{inversion}'").reset_index()
+
+    @check_types
+    def karyotype(
+        self,
+        inversion: inversion_param,
+        sample_sets: Optional[base_params.sample_sets] = None,
+        sample_query: Optional[base_params.sample_query] = None,
+    ) -> pd.DataFrame:
+        """
+        Infer karyotype from tag SNPs for a given inversion.
+
+        Parameters
+        ----------
+        inversion : str
+            Name of inversion to infer karyotype for. One of ["2La", "2Rb", "2Rc_gam", "2Rc_col", "2Rd", "2Rj"].
+        sample_sets : list of str, optional
+            Sample sets to include in the analysis. If None (default), use all sample sets.
+        sample_query : str, optional
+            Query to filter samples. If None (default), use all samples.
+        """
```

Comment:
Would you mind migrating this to use the `@doc(...)` decorator? Should mean you only then need to provide the summary, and the parameter descriptions will get pulled in automatically.